### DISCUSSION TITLE
disk space utilisation behaviour, space_reservation

### DISCUSSION THREAD

**Discussion by karl-hape** [abraunegg/onedrive#2925] [General]
Good evening,

I set-up Onedrive onedrive v2.5.2-1+np1+1.1 on Ubuntu 22.04.5 LTS.
All is working well but I have a question on local disk usage and the config option space_reservation.

This answer in the user manual to this question
https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#how-can-i-prevent-my-local-disk-from-filling-up
states that "the application will reserve 50 MB of disk space to prevent your filesystem from running out of disk space."

Same as at:
https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#overview-of-all-onedrive-client-for-linux-cli-options
```
  --space-reservation ARG
      The amount of disk space to reserve (in MB) to avoid 100% disk space utilisation
```
Therefore, my assumption was that if I put a file of e.g. 3 GB in the OneDrive Folder, the program will upload it, start to free up local disk space and eventually come back to max. 50 MB of local disk space usage.
"local disk usage < online usage"

However, at this discussion:
https://github.com/abraunegg/onedrive/discussions/1931
The answer to a similar question is "the data will be available when you are offline as this is making a local copy of your online data."

My personal observation supports this:
```
$ du -h OneDrive/
3,3G    OneDrive/
```
Nautilus (standard ubuntu file manager) shows the same.

This indicates a behaviour: always "local disk usage = online usage". 

I uncommented and varied the value in the configuration file to be sure that the value is taken over which works fine: 
space_reservation = "50"
```
$ onedrive --display-config
Config option 'space_reservation'            = 52428800
```

Would it be possible for you to share some explanations on this and the acutal intendend behaviour?
Is it possible to achieve the behaviour "local disk usage < online usage" ? (Basically the "free up space" behaviour on windows.)

Thank you.

**Comment by abraunegg** [comment]
> Would it be possible for you to share some explanations on this and the acutal intendend behaviour?

As per the documentation it is the amount free space on your local disk to leave so that your local disk does not run out of disk space - as if you run out of disk space locally - you cant even log into your system .. can cause corruption, and all manner of other issues.

> Therefore, my assumption was that if I put a file of e.g. 3 GB in the OneDrive Folder, the program will upload it, start to free up local disk space and eventually come back to max. 50 MB of local disk space usage.

This assumption is 100% wrong.

Your local filesystem - your PC - has a 'disk' where you are storing files. This option has nothing to do with online free space or anything to do with OneDrive itself - but to protect yourself from writing too much data locally, filling up your disk, so that your system is rendered unusable.

> Is it possible to achieve the behaviour "local disk usage < online usage" ? (Basically the "free up space" behaviour on windows.)

Please read the documentation regarding the application options:
* https://github.com/abraunegg/onedrive/blob/master/docs/application-config-options.md#remove_source_files

**Reply by karl-hape** [reply]
Thanks for the clarification. It did not appear to me that this config refers to the remaining amount of disk space to reserve. But it's clear now.
